assets/layout/components/BaseApp.jsx

```diff
@@ -96,8 +96,9 @@ export default class BaseApp extends React.Component {
         }
     }
 
-    filterActions(item) {
-        return this.props.actions.filter((action) => !action.when || action.when(this.props.state, item));
+    filterActions(item, config) {
```

Comment:
This can probably be a utils function as it's definition repeated many times.

Reply:
You're right Vikas, I'll merge for now but will create another PR for the refactoring.